src/Functions/FunctionDateOrDateTimeToSomething.h

```diff
@@ -79,7 +28,7 @@ class FunctionDateOrDateTimeToSomething : public IFunction
             /// to accommodate functions like toStartOfDay(today()), toStartOfDay(yesterday()) etc.
             if (arguments.size() == 2 && time_zone.empty())
                 throw Exception(
-                    "Function " + getName() + " supports a 2nd argument (optional) that must be non-empty and be a valid time zone",
+                    "Function " + this->getName() + " supports a 2nd argument (optional) that must be non-empty and be a valid time zone",
```

Comment:
The code in the `if constexpr` block in l. 35 ff. is not found in `FunctionDateOrDateTimeToDateOrDate32::getReturnTypeImpl()`. Intended?

Reply:
Yes, I confirm, it's correct. These checks are not needed in `FunctionDateOrDateTimeToDateOrDate32::getReturnTypeImpl().`